sections/futures/MarketInfoBox/MarketInfoBox.tsx

```diff
@@ -25,78 +25,88 @@ const MarketInfoBox: React.FC = () => {
 	const maxLeverage = useAppSelector(selectMaxLeverage);
 	const { commitDeposit } = useAppSelector(selectDelayedOrderFee);
 
-	const totalMargin = position?.remainingMargin ?? zeroBN;
-	const availableMargin = position?.accessibleMargin ?? zeroBN;
-
-	const buyingPower = totalMargin.gt(zeroBN) ? totalMargin.mul(maxLeverage ?? zeroBN) : zeroBN;
-
-	const marginUsage = availableMargin.gt(zeroBN)
-		? totalMargin.sub(availableMargin).div(totalMargin)
-		: zeroBN;
-
 	const minInitialMargin = useMemo(() => marketInfo?.minInitialMargin ?? zeroBN, [
 		marketInfo?.minInitialMargin,
 	]);
 
-	const isDelayedOrder = useMemo(() => orderType === 'delayed', [orderType]);
-
-	const totalDeposit = useMemo(() => {
-		return (commitDeposit ?? zeroBN).add(marketInfo?.keeperDeposit ?? zeroBN);
-	}, [commitDeposit, marketInfo?.keeperDeposit]);
+	const totalMargin = position?.remainingMargin ?? zeroBN;
 
-	const getPotentialAvailableMargin = useCallback(
-		(trade: FuturesPotentialTradeDetails | null, marketMaxLeverage?: Wei) => {
-			let inaccessible =
-				(marketMaxLeverage && trade?.notionalValue.div(marketMaxLeverage).abs()) ?? zeroBN;
+	// function for calculating available margin
+	const getAvailableMargin = useCallback(
+		(notionalValue: Wei, margin: Wei, marketMaxLeverage: Wei) => {
+			let inaccessible = notionalValue.div(marketMaxLeverage).abs() ?? zeroBN;
 
 			// If the user has a position open, we'll enforce a min initial margin requirement.
 			if (inaccessible.gt(0) && inaccessible.lt(minInitialMargin)) {
 				inaccessible = minInitialMargin;
 			}
 
 			// check if available margin will be less than 0
-			return trade?.margin?.sub(inaccessible).gt(0)
-				? trade?.margin?.sub(inaccessible).abs()
-				: zeroBN;
+			return margin.sub(inaccessible).gt(0) ? margin.sub(inaccessible).abs() : zeroBN;
 		},
 		[minInitialMargin]
 	);
 
-	const previewAvailableMargin = React.useMemo(() => {
-		const potentialAvailableMargin = getPotentialAvailableMargin(
-			potentialTrade,
-			marketInfo?.maxLeverage
-		);
+	// adjust accessible margin due to frontend soft cap on leverage
+	const availableMargin = useMemo(() => {
+		if (!position?.position || !marketInfo) return zeroBN;
+		return getAvailableMargin(position.position.notionalValue, totalMargin, marketInfo.maxLeverage);
```

Comment:
Should this also be using the selector `maxLeverage` with our local cap rather than the market leverage?

Reply:
No, because the `getAvailableMargin` function uses the actual static max leverage for the calculation. It's something like `total margin - (notional size / max leverage)`.

If we use the selector, it changes depending on the trade size, so this function can't be reused for both calculating "available margin" and "preview available margin".

Also - The reason why I'm using a function to calculate "available margin" here instead of using `accessibleMargin` is because the contract returns a value valid for 100x leverage. Using this function will account for our settings override. Same was true of "buying power" since it was calculated from "available margin"

Reply:
ok makes sense, I was mostly wondering if `marketInfo.maxLeverage` is 100x but I guess it's not.